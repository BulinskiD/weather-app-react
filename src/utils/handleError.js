export default (error, setLoading) => {
    console.log(error);
    let errorMessage;
    switch(error.response.status) {
        case 404:
            errorMessage = "Nie znaleziono miasta";
            break;
        case 500:
            errorMessage = "Serwer jest chwilowo niedostępny, spróbuj później!"
            break;
        default:
            errorMessage = "Coś poszło nie tak..."
    }
    setLoading(false);
    return errorMessage;
}
