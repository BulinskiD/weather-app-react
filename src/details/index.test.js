import React from 'react';
import mockAxios from 'jest-mock-axios';
import Details from './index';
import handleError from '../utils/handleError';
import {shallow, mount} from "enzyme";
import {act} from "react-dom/test-utils";

jest.mock('../utils/handleError');

handleError.mockReturnValue("Test");

afterEach(()=>{
    mockAxios.reset();
    handleError.mockClear();
});

describe('Details', ()=>{

    const id = {match: {params:{id: 1}}};
    const unit = "imperial";

    const axiosResponseOK = {city: {coord: {lat: 1, lon: 1}}, list: []};
    const axiosResponseError = {
        data: {},
        status: 500,
        statusText: 'Internal server error',
        headers: {},
        config: {},
    };

    it('should call axios on load', ()=>{
        let component;
        act(()=>{
            component= mount(<Details {...id} unit={unit}/>);
        });
        mockAxios.mockResponse(axiosResponseOK);
        expect(component).toMatchSnapshot();
        expect(mockAxios.get).toBeCalledTimes(1);
    });

    it('should set error message properly when request fails on load', ()=>{
        act(()=>{
            mount(<Details {...id} unit={unit}/>);
        });
        try {
            mockAxios.mockError(axiosResponseError);
        } catch(error) {
            expect(mockAxios.get).toBeCalledTimes(1);
            expect(handleError).toBeCalledWith(axiosResponseError);
        }
    });

});
