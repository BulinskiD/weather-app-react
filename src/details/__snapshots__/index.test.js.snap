// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should render correctly with given props 1`] = `""`;
